apps/prairielearn/src/middlewares/workspaceProxy.ts

```diff
@@ -130,16 +131,17 @@ export function makeWorkspaceProxyMiddleware() {
       if (!match) throw new Error(`Could not match path: ${path}`);
 
       const workspace_id = match[1];
-      const result = await queryZeroOrOneRowAsync(
+      const result = await queryOptionalRow(
         "SELECT hostname FROM workspaces WHERE id = $workspace_id AND state = 'running';",
         { workspace_id },
+        z.object({ hostname: z.string() }),
```

Comment:
Besides the single-column comment above, we often use a construct like:
```suggestion
        WorkspaceSchema.shape.hostname,
```
This ensures we match the column type with the schema type, and any changes in the type are properly reflected.